pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -407,12 +407,12 @@ func (p *DNSProxy) SetRejectReply(opt string) {
 	}
 }
 
-// ExtractMsgDetails extracts a canonical query name, any IPs in a response and
-// the lowest applicable TTL. When a CNAME is returned the chain is collapsed
-// down, keeping the lowest TTL, and CNAME targets are returned.
-func ExtractMsgDetails(msg *dns.Msg) (qname string, responseIPs []net.IP, TTL uint32, CNAMEs []string, err error) {
+// ExtractMsgDetails extracts a canonical query name, any IPs in a response,
+// the lowest applicable TTL and message type. When a CNAME is returned the
+// chain is collapsed down, keeping the lowest TTL, and CNAME targets are returned.
+func ExtractMsgDetails(msg *dns.Msg) (qname string, responseIPs []net.IP, TTL uint32, CNAMEs []string, msgType int, recordTypes []uint16, err error) {
```

Comment:
Maybe leave a comment here explaining that while it is a uint16, recordType is intended to reflect the RRType in the dns package.